### PR TITLE
Hometask #1

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

**Comment by qwwdfsad** [comment]
Коммит с тестом на wait опоздал к дедлайну на 1 минуту по моим часам :(

File: 23/1/background.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+rm -f out.txt 2> /dev/null
+echo Starting 3 background scripts (5 sec sleep)
+./echo_and_sleep.sh &
+./echo_and_sleep.sh &
+./echo_and_sleep.sh &
+echo Starting same script synchronously (5 sec sleep)
+./echo_and_sleep.sh
+echo Synchronous script finished
+echo Printing lines count, 4 expected:
+wc -l out.txt
```
Comment (krinkin) [inline 23/1/background.sh:10]: add CRLN

File: 23/1/cmd.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ls -lah > y.txt
+cat < y.txt | sort | uniq | wc -l > y1.txt
+cat y1.txt
+rm y1.txt
+ls | sort | uniq | wc
+rm y.txt
```
Comment (krinkin) [inline 23/1/cmd.sh:6]: CRLN

File: 23/1/sh.c [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+    struct pipecmd *pcmd;
+    struct redircmd *rcmd;
+
+    if (cmd == 0) {
+        exit(0);
+    }
+
+    switch (cmd->type) {
+        default: {
+            fprintf(stderr, "Unknown runcmd\n");
```
Comment (krinkin) [inline 23/1/sh.c:59]: see also perror(3)

File: 23/1/sh.c [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+int tryParsePid(char *string);
+
+void wait_for_child(char *buf, int *background_pids);
+
+void register_background_child(int *background_pids, int *current_counter, int pid);
+
+void process_background_children(int *background_pids);
+
+int main(void) {
+    static char buf[100];
```
Comment (krinkin) [inline 23/1/sh.c:155]: 100?

**Review by krinkin — APPROVED** [review]
Good job!

BTW, add CRLF in all files as a rule;

File: 23/1/sh.c (hunk 4 above)
Comment (qwwdfsad) [reply 23/1/sh.c:155]: It is a constant from original sh.c.
It equals to MAX_BACKGROUND_PROCESS_TO_TRACK accidentally and has nothing in common with it

**Pull request merged**